src/stores/ProductsListStore/ProductsListStore.ts

```diff
@@ -0,0 +1,224 @@
+    // Первый аргумент – колбэк, который возвращает отслеживаемые поля
+    () => this.filterStore.value,
+
+    // Второй аргумент – колбэк, в котором выполняется желаемая логика
+    (newValue) => {
+      console.log('reactionFiltr -' + newValue);
+      this.fetchProducts(1, '', Array.from(this.filterStore.selectedKeysSet));
+    },
+  );
+
+
+
+
+  constructor() {
+    makeObservable<ProductsListStore, PrivateFields>(this, {
+      _products: observable.ref,
+      _meta: observable,
+      products: computed,
+      meta: computed,
+      totalProducts: computed,
+      fetchProducts: action,
+    });
+
+    // Устанавливаем начальные значения из query параметров
+    // const initialParams = rootStore.query.params;
+
+    this.searchStore.setSearchValue((rootStore.query.getParam('search') as string) || '');
+    this.paginationStore.setCurrentPage(parseInt(rootStore.query.getParam('page') as string) || 1);
+
+    //Реакция на изменения текущей страницы
+    reaction(
+      () => this.paginationStore.currentPage,
+      (currentPage) => {
+        console.log(
+          'Page reaction ',
+          currentPage,
+          ' Qs ',
+          qs.stringify({
+            page: currentPage,
+          }),
+        );
+
+        // rootStore.query._params['page'] = currentPage;
+        const newUrl = `${window.location.pathname}?${qs.stringify({
+          search: rootStore.query.getParam('search') || '',
+          filter: rootStore.query.getParam('filter') || '', // Сохраняем массив как строку
+          page: currentPage,
+        })}`;
+        window.history.replaceState(null, '', newUrl);
+
+        console.log('тут вызывается setSerch?');
+
+      },
+    );
+
+    // Реакция на изменения поиска
+    reaction(
+      () => this.searchStore.searchValue,
+      (newSearchValue) => {
+        console.log('search value change', newSearchValue);
+
+        const newUrl = `${window.location.pathname}?${qs.stringify({
+          search: newSearchValue,
+          filter: rootStore.query.getParam('filter') || '', // Сохраняем массив как строку
+          page: 1,
+        })}`;
+        window.history.replaceState(null, '', newUrl);
+
+      },
+    );
+
+    // Реакция на изменения фильтров
+    reaction(
+      () => Array.from(this.filterStore.selectedKeysSet),
+      (newFilter) => {
+        const newUrl = `${window.location.pathname}?${qs.stringify({
+          search: rootStore.query.getParam('search') || '',
+          filter: newFilter.join(','),
+          page: 1,
+        })}`;
+        window.history.replaceState(null, '', newUrl);
+      },
+    );
+  }
+
+  // TODO экземпляры моделей поиска и пагинации создавать в сторе страницы списка и оттуда брать и подставлять значения в запрос
+
+  get products(): Map<number, ProductInfoModel[]> {
+    return this._products;
+  }
+
+  get meta(): Meta {
+    return this._meta;
+  }
+
+  get totalProducts(): number {
+    return this._totalProducts;
+  }
```

Comment:
Думаю поле this._totalProducts вообще не нужно
```suggestion
  get totalProducts(): number {
    return this.products.length;
  }
```